src/storage-client/src/sink.rs

```diff
@@ -336,17 +344,16 @@ pub async fn build_kafka(
     }
 
     match &connection.consistency_config {
-        KafkaConsistencyConfig::Progress { topic } => {
-            ensure_kafka_topic(
-                &client,
-                topic,
-                1,
-                connection.replication_factor,
-                KafkaSinkConnectionRetention::default(),
-            )
-            .await
-            .context("error registering kafka consistency topic for sink")?;
-        }
+        KafkaConsistencyConfig::Progress { topic } => ensure_kafka_topic(
+            &client,
+            topic,
+            1,
+            connection.replication_factor,
+            KafkaSinkConnectionRetention::default(),
+        )
+        .await
+        .check_ssh_status(client.inner().context())
+        .add_context("error registering kafka consistency topic for sink")?,
```

Comment:
This doesn't look terribly unmanageable, but it does seem easy for someone for someone to refactor and lose a `check_ssh_status` call, or add a new call to a Kafka client method and forget to call `check_ssh_status`. I don't have anything better to propose, though.

Reply:
Yeah...i couldn't think of anything better...